src/main/kotlin/com/faforever/userservice/ui/view/registration/ActivateView.kt

```diff
@@ -0,0 +1,126 @@
+import com.vaadin.flow.router.BeforeEnterEvent
+import com.vaadin.flow.router.BeforeEnterObserver
+import com.vaadin.flow.router.Route
+import com.vaadin.flow.server.VaadinSession
+
+@Route("/register/activate", layout = CardLayout::class)
+class ActivateView(private val registrationService: RegistrationService) :
+    CompactVerticalLayout(), BeforeEnterObserver {
+
+    class PasswordConfirmation {
+        var password: String = ""
+        var confirmedPassword: String = ""
+    }
+
+    private val username = TextField(null, getTranslation("register.username")).apply {
+        isReadOnly = true
+        setWidthFull()
+    }
+    private val email = TextField(null, getTranslation("register.email")).apply {
+        isReadOnly = true
+        setWidthFull()
+    }
+    private val password = PasswordField(null, getTranslation("register.password")).apply {
+        setWidthFull()
+        valueChangeMode = ValueChangeMode.LAZY
+    }
+    private val confirmedPassword = PasswordField(null, getTranslation("register.password.confirm")).apply {
+        setWidthFull()
+        valueChangeMode = ValueChangeMode.LAZY
+    }
+    private val submit = Button(getTranslation("register.activate")) { activate() }.apply {
+        isEnabled = false
+        setWidthFull()
+        addThemeVariants(ButtonVariant.LUMO_PRIMARY)
+    }
+
+    private val binder = Binder(PasswordConfirmation::class.java)
+
+    private lateinit var registeredUser: RegisteredUser
+
+    init {
+
+        val formHeaderLeft = FafLogo()
+        val formHeaderRight = H2(getTranslation("register.activate"))
+        val formHeader = HorizontalLayout(formHeaderLeft, formHeaderRight).apply {
+            justifyContentMode = FlexComponent.JustifyContentMode.CENTER
+            alignItems = FlexComponent.Alignment.CENTER
+            setId("form-header")
+            setWidthFull()
+        }
+
+        add(formHeader)
+
+        add(username, email, password, confirmedPassword, submit)
+
+        val footer = VerticalLayout(SocialIcons()).apply {
+            alignItems = FlexComponent.Alignment.CENTER
+        }
+
+        add(footer)
+
+        binder.forField(password)
+            .asRequired(getTranslation("register.password.required"))
+            .withValidator({ username -> username.length >= 6 }, getTranslation("register.password.size"))
+            .bind("password")
+
+        binder.forField(confirmedPassword)
+            .withValidator(
+                { confirmedPassword -> confirmedPassword == password.value },
+                getTranslation("register.password.match"),
+            ).bind("confirmedPassword")
+
+        binder.addStatusChangeListener { submit.isEnabled = it.binder.isValid }
+    }
+
+    private fun activate() {
+        val validationStatus = binder.validate()
+        if (validationStatus.hasErrors()) {
+            return
+        }
+
+        val ipAddress = IpAddress(VaadinSession.getCurrent().browser.address)
```

Comment:
Does this take proxy/cloudflare headers into account?

Reply:
No it doesn't so need to update that